src/lib/y2partitioner/device_graphs.rb

```diff
@@ -11,37 +11,22 @@ class DeviceGraphs
     # Run a block with a **duplicate** of @dg.
     # If the block returns a truthy value,
     # the new dg is copied to the old one
-    # (which is useful if the orig one was the `staging` one)
+    # (which is useful if the orig one was the `current` one)
```

Comment:
Adding here general comments for this class. It was agreed to accept it in the previous PR just in temporary form. So I guess now is the correct time to complain about stuff. :wink:

1. I think it's a bad idea to use `DeviceGraphs` instead of `Devicegraphs`. Libstorage-ng uses the later capitalization and yast2-storage-ng respects that decision. Introducing here a different capitalization for classes referring to the same concept can only bring errors.

2. I miss better documentation. Specially about the attributes `current` and `original`.
    1. `original` does not represent the starting point of the partitioner, but the devicegraph that is considered to represent the system. That is, the partitioner can also be started taking an already proposed devicegraph as starting point, but even in this case, `original` is not that initial proposed devicegraph, but the system one. That's not obvious from the names.

    2. The current implementation implies that `current` will be directly modified, which leaves the responsibility of doing the initial `staging.dup` to the caller. That must be very clearly stated, since it's not the default Ruby behavior.  That's why I proposed this alternative approach[1], in which the constructor just receives the initial devicegraph and takes care of copying it into the working/result devicegraph. So it's closer to the default Ruby way of not modifying the arguments (that is, no input/output arguments). I'm not saying we have to change this, but we have to document it since it's dangerous and we are not protecting the user of the class to shoot its own foot.

[1] https://github.com/yast/yast-partitioner/pull/6/files#r121332203